### DISCUSSION TITLE
Make the artisan down command work on multiple servers

### DISCUSSION THREAD

**Discussion by ibrunotome** [laravel/framework#36474] [Ideas] [answered]
This was proposed [here](https://github.com/laravel/ideas/issues/1944).

The problem: when you have multiple Kubernetes pods (or multiple application servers), the down command will only write the file on one disk, but if the application is served from another pod or server, it will be running normally.

The option to write the data on redis or database will solve this problem. Here is one example overriding the application to get this working https://www.raypold.com/posts/laravel-distributed-maintenance-mode/

**Comment by wimulkeman** [comment, marked as answer]
I've created a pull request regarding this discussion https://github.com/laravel/framework/pull/40059. It should resolve the possibility to easily replace the maintenance mode logic.

The only point it doesn't address is the maintenance.php file which could be placed under storage/framework directory. I don't known how much it is used but requires some special attention for it can be called before Laravel is initialised.
